dbms/src/Functions/FunctionsJson.h

```diff
@@ -1515,6 +1515,375 @@ class FunctionJsonDepth : public IFunction
     }
 };
 
+class FunctionJsonContainsPath : public IFunction
+{
+public:
+    static constexpr auto name = "json_contains_path";
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionJsonContainsPath>(); }
+
+    String getName() const override { return name; }
+
+    bool isVariadic() const override { return true; }
+
+    size_t getNumberOfArguments() const override { return 0; }
+
+    bool useDefaultImplementationForNulls() const override { return false; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if unlikely (arguments.size() < 3)
+        {
+            throw Exception(
+                fmt::format("Illegal arguments count {} of function {}", arguments.size(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+        for (const auto & arg : arguments)
+        {
+            if unlikely (!arg->onlyNull() && !removeNullable(arg)->isString())
+            {
+                throw Exception(
+                    fmt::format("Illegal type {} of argument of function {}", arg->getName(), getName()),
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+            }
+        }
+        if (arguments[0]->onlyNull() || arguments[1]->onlyNull() || arguments[2]->onlyNull())
+            return makeNullable(std::make_shared<DataTypeNothing>());
+        else
+        {
+            auto return_type = std::make_shared<DataTypeUInt8>();
+            for (const auto & arg : arguments)
+            {
+                if (arg->onlyNull() || arg->isNullable())
+                    return makeNullable(return_type);
+            }
+            return return_type;
+        }
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        const auto & json_col = block.getByPosition(arguments[0]).column;
+        const auto & type_col = block.getByPosition(arguments[1]).column;
+        if (json_col->onlyNull() || type_col->onlyNull() || block.getByPosition(arguments[2]).column->onlyNull())
```

Comment:
https://github.com/pingcap/tidb/blob/11aefee7211a85ff6ec4c2fb1eea0897b84c5562/pkg/expression/builtin_json.go#L687-L694
Because even if args[3] is onlyNull, the result may not be onlyNull.
But if args[2] is onlyNull, the result must be onlyNull.

Reply:
It's strange. However, mysql behaves like this, looks like a mysql bug, and tidb studied...